src/vulkan-renderer/wrapper/device.cpp

```diff
@@ -0,0 +1,220 @@
+#include "inexor/vulkan-renderer/wrapper/device.hpp"
+
+namespace {
+// TODO: Make proper use of queue priorities in the future.
+constexpr float default_queue_priority = 1.0f;
+} // namespace
+
+namespace inexor::vulkan_renderer::wrapper {
+
+Device::Device(Device &&other) noexcept : device(other.device), graphics_card(other.graphics_card) {}
+
+Device::Device(const VkInstance instance, const VkSurfaceKHR surface, bool enable_vulkan_debug_markers, bool prefer_distinct_transfer_queue,
+               const std::optional<std::uint32_t> preferred_physical_device_index)
+    : graphics_card(graphics_card), surface(surface) {
+
+    std::optional<VkPhysicalDevice> selected_graphics_card =
+        settings_decision_maker.decide_which_graphics_card_to_use(instance, surface, preferred_physical_device_index);
+
+    if (!selected_graphics_card) {
+        throw std::runtime_error("Error: Could not find suitable graphics card!");
+    }
+
+    graphics_card = selected_graphics_card.value();
+
+    spdlog::debug("Creating Vulkan device queues.");
+
+    std::vector<VkDeviceQueueCreateInfo> queues_to_create;
+
+    if (prefer_distinct_transfer_queue) {
+        spdlog::debug("The application will try to use a distinct data transfer queue if it is available.");
+    } else {
+        spdlog::warn("The application is forced not to use a distinct data transfer queue!");
+    }
+
+    // Check if there is one queue family which can be used for both graphics and presentation.
+    std::optional<std::uint32_t> queue_family_index_for_both_graphics_and_presentation =
+        settings_decision_maker.find_queue_family_for_both_graphics_and_presentation(graphics_card, surface);
+
+    bool use_distinct_data_transfer_queue = false;
```

Comment:
Value is never be used.

Reply:
It is, but many lines after declaration. I moved it to a better place.